plugin-server/src/worker/vm/promise-manager.ts

```diff
@@ -13,16 +14,19 @@ export class PromiseManager {
         this.statsd = statsd
     }
 
-    public trackPromise(promise: Promise<any>): void {
+    public trackPromise(promise: Promise<any>, key: string): void {
         if (typeof promise === 'undefined') {
             return
         }
 
+        status.info('🤝', `Tracking promise ${key}`)
+        this.statsd?.increment(`worker_promise_manager_promise_start`, { key })
         this.pendingPromises.add(promise)
 
         promise.finally(() => {
             this.pendingPromises.delete(promise)
         })
+        this.statsd?.increment(`worker_promise_manager_promise_end`, { key })
     }
 
     public async awaitPromisesIfNeeded(): Promise<void> {
```

Comment:
This is the bit I think it would be useful to have some logging in. Like how long do we end up waiting on the function.

Reply:
yes ... good point, will update